src/conv2d.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+import {Tensor} from './lib/tensor.js';
+import {validateConv2dParams} from './lib/validate-input.js';
+import {computePaddingForAutoPad} from './lib/compute-padding.js';
+import {transpose} from './transpose.js';
+
+/**
+ * Compute a 2-D convolution given 4-D input and filter tensors.
+ * @param {Tensor} input
+ * @param {Tensor} filter
+ * @param {MLConv2dOptions} options
+ * @return {Tensor}
+ */
+export function conv2d(input, filter, {padding = [0, 0, 0, 0],
```

Comment:
@wchao1115, this function implements the steps of conv2d, just for your convivence, thanks!